src/components/riveMonster/rive-monster-component.ts

```diff
@@ -1,70 +1,154 @@
 
 interface RiveMonsterComponentProps {
-    canvas: HTMLCanvasElement;  // Canvas element where the animation will render
-    autoplay: boolean;
-    fit?: string;  // Fit property (e.g contain, cover, etc.)
-    alignment?: string;  // Alignment property (e.g topCenter, bottomLeft, etc.)
-    width?: number;  // Optional width for the Rive animation
-    height?: number;  // Optional height for the Rive animation
-    onLoad?: () => void;  // Callback once Rive animation is loaded
+  canvas: HTMLCanvasElement; // Canvas element where the animation will render
+  autoplay: boolean;
+  fit?: string; // Fit property (e.g contain, cover, etc.)
+  alignment?: string; // Alignment property (e.g topCenter, bottomLeft, etc.)
+  width?: number; // Optional width for the Rive animation
+  height?: number; // Optional height for the Rive animation
+  onLoad?: () => void; // Callback once Rive animation is loaded
+  gameCanvas?: HTMLCanvasElement; // Main canvas element
 }
 
 export class RiveMonsterComponent {
-    private props: RiveMonsterComponentProps;
-    private riveInstance: any;
-    private src: string = './assets/monsterrive.riv';  // Define the .riv file path
-    private stateMachines: string = 'State Machine 1';  // Define the state machine
-
-    // Static readonly properties for all monster animations
-    public static readonly Animations = {
-        OPENING_MOUTH_EAT: 'Opening Mouth Eat',
-        EAT_HAPPY: 'Eat Happy',
-        IDLE: 'Idle',
-        EAT_DISGUST: 'Eat Disgust',
-    };
-
-
-    constructor(props: RiveMonsterComponentProps) {
-        this.props = props;
-
-        
-        // Initialize Rive
-        this.riveInstance = new Rive({
-            src: this.src,
-            canvas: this.props.canvas,
-            autoplay: this.props.autoplay,
-            stateMachines: this.stateMachines,
-            layout: new Layout({
-                fit: Fit[this.props.fit || "Contain"],
-                alignment: Alignment[this.props.alignment || "TopCenter"],
-            }),
-            onLoad: () => {
-                if (this.props.onLoad) {
-                    this.props.onLoad();
-                }
-            },
-        });
+  private props: RiveMonsterComponentProps;
+  private riveInstance: any;
+  private src: string = './assets/monsterrive.riv'; // Define the .riv file path
+  private stateMachines: string = 'State Machine 1'; // Define the state machine
+  public game: any;
+  public x: number;
+  public y: number;
+  private hitboxRangeX: {
+    from: number;
+    to: number;
+  };
+  private hitboxRangeY: {
+    from: number;
+    to: number;
+  };
+
+  // Static readonly properties for all monster animations
+  public static readonly Animations = {
+    OPENING_MOUTH_EAT: 'Opening Mouth Eat',
+    EAT_HAPPY: 'Eat Happy',
+    IDLE: 'Idle',
+    EAT_DISGUST: 'Eat Disgust',
+  };
+
+  constructor(props: RiveMonsterComponentProps) {
+    this.props = props;
+
+    if (this.props.gameCanvas) {
```

Comment:
Add a comment regarding the this.props.gameCanvas will not be needed once FM-333 pixelate issue fix has been applied.

Reply:
Thanks added a todo comment